### DISCUSSION TITLE
Delta Temporality Metrics with the C++ SDK / Collector

### DISCUSSION THREAD

**Discussion by Michaelh12304-work** [open-telemetry/opentelemetry-cpp#2589] [Q&A]
Is there a way to produce Metrics with the Delta Temporality  using the C++ SDK / Collector??

I see this env var for example:  OTEL_EXPORTER_OTLP_METRICS_TEMPORALITY_PREFERENCE   which does not seem to be present in the C++ SDK source code.

Some APM providers, in fact a number do not like the non delta kinds of metrics.

We are trying to use the doubleUpDownCounter and the Counter<double> -- which is the one that does not work with Dynatrace.

Thanks

**Comment by lalitb** [comment]
@Michaelh12304-work  You can set the preferred temporality as part of OTLP grpc/http exporter options - 
https://github.com/open-telemetry/opentelemetry-cpp/blob/19f0dc7760e2852efacd1ba9fe5b4cbfbf3edfb2/exporters/otlp/include/opentelemetry/exporters/otlp/otlp_grpc_metric_exporter_options.h#L31


So something like this should work (rough code to give an idea):
```cpp
OtlpGrpcMetricExporterOptions  options;
options.aggregation_temporality = PreferredAggregationTemporality::kDelta // or kCumulative or kLowMemory
OtlpGrpcMetricExporter exporter(options);
```

where (as per the [specs](https://github.com/open-telemetry/opentelemetry-specification/blob/efd1c9840273a8420edbd80e46df4cf72191b9ac/specification/metrics/sdk_exporters/otlp.md#additional-configuration)),
* `Cumulative`: Choose cumulative aggregation temporality for all instrument kinds.
* `Delta`: Choose Delta aggregation temporality for Counter, Asynchronous Counter and Histogram instrument kinds, choose
  Cumulative aggregation for UpDownCounter and Asynchronous UpDownCounter instrument kinds.
* `LowMemory`: This configuration uses Delta aggregation temporality for Synchronous Counter and Histogram and uses Cumulative aggregation temporality for Synchronous UpDownCounter, Asynchronous Counter, and Asynchronous UpDownCounter instrument kinds.
